pyntc/devices/nxos_device.py

```diff
@@ -37,35 +39,43 @@ def __init__(self, host, username, password, transport="http", timeout=30, port=
             transport (str, optional): Transport protocol to connect to device. Defaults to "http".
             timeout (int, optional): Timeout in seconds. Defaults to 30.
             port (int, optional): Port used to connect to device. Defaults to None.
+            verify (bool, optional): SSL verification.
+            kwargs: Left for compatibility with other tools, for instance nautobot-inventory may pass additional kwargs.
+
         """
         super().__init__(host, username, password, device_type="cisco_nxos_nxapi")
         self.transport = transport
         self.timeout = timeout
-        self.native = NXOSNative(host, username, password, transport=transport, timeout=timeout, port=port, **kwargs)
+        self.native = NXOSNative(
+            host, username, password, transport=transport, timeout=timeout, port=port, verify=verify
+        )
         log.init(host=host)
 
     def _image_booted(self, image_name, **vendor_specifics):
         version_data = self.show("show version", raw_text=True)
-        if re.search(image_name, version_data):
-            log.info("Host %s: Image %s booted successfully.", self.host, image_name)
-            return True
-
-        log.info("Host %s: Image %s not booted successfully.", self.host, image_name)
-        return False
+        return bool(re.search(image_name, version_data))
 
     def _wait_for_device_reboot(self, timeout=3600):
         start = time.time()
         while time.time() - start < timeout:
-            try:
-                self.native.show("show hostname")
-                log.debug("Host %s: Device rebooted.", self.host)
-                return
+            try:  # NXOS stays online, when it installs OS
+                self.refresh()
+                if self.uptime < 180:
+                    log.info("Host %s: Device rebooted.", self.host)
+                    return
```

Comment:
This has been changed to look for uptime, because NXOS stays on-line while it installs new image.
To make that working, refresh() method has been implemented with proper fact cache clear.